src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Escape.cs

```diff
@@ -0,0 +1,185 @@
+        /// within an expression parsed with x mode. If future Regex features add
+        /// additional metacharacters, developers should depend on Escape to escape those
+        /// characters as well.)
+        /// </summary>
+        /// <returns>Returns the amount of chars written into the output Span.</returns>
+        public static bool TryEscape(ReadOnlySpan<char> str, Span<char> destination, out int charsWritten)
+        {
+            Escape(targetSpan: true, str, destination, out charsWritten, out bool spanSuccess);
+
+            return spanSuccess;
+        }
+
+        /// <summary>
+        /// Unescapes any escaped characters in the input string.
+        /// </summary>
+        public static string Unescape(string str)
+        {
+            if (str == null)
+                throw new ArgumentNullException(nameof(str));
+
+            return Unescape(targetSpan: false, str.AsSpan(), Span<char>.Empty, out _, out _);
+        }
+
+        /// <summary>
+        /// Unescapes any escaped characters in the input text.
+        /// </summary>
+        public static bool TryUnescape(ReadOnlySpan<char> str, Span<char> destination, out int charsWritten)
+        {
+            Unescape(targetSpan: true, str, destination, out charsWritten, out bool spanSuccess);
+
+            return spanSuccess;
+        }
+
+        /// <summary>
+        /// Escapes all metacharacters (including |,(,),[,{,|,^,$,*,+,?,\, spaces and #)
+        /// </summary>
+        private static string Escape(bool targetSpan, ReadOnlySpan<char> input, Span<char> destination, out int charsWritten, out bool spanSuccess)
+        {
+            for (int i = 0; i < input.Length; i++)
+            {
+                if (RegexCharCategory.IsMetachar(input[i]))
+                {
+                    return EscapeImpl(targetSpan, input, i, destination, out charsWritten, out spanSuccess);
+                }
+            }
+
+            // If nothing to escape, return the input.
+            return input.CopyInput(targetSpan, destination, out charsWritten, out spanSuccess);
+        }
+
+        private static string EscapeImpl(bool targetSpan, ReadOnlySpan<char> input, int i, Span<char> destination, out int charsWritten, out bool spanSuccess)
+        {
+            // For small inputs we allocate on the stack. In most cases a buffer three 
+            // times larger the original string should be sufficient as usually not all 
+            // characters need to be encoded.
+            // For larger string we rent the input string's length plus a fixed 
+            // conservative amount of chars from the ArrayPool.
+            Span<char> buffer = input.Length <= (EscapeMaxBufferSize / 3) ? stackalloc char[EscapeMaxBufferSize] : default;
+            ValueStringBuilder vsb = !buffer.IsEmpty ?
+                new ValueStringBuilder(buffer) :
+                new ValueStringBuilder(input.Length + 200);
```

Comment:
You didn't change this code, but I wonder how we chose to use addition here. If I had to guess how much longer a string would get after escaping, I would expect on average it would increase by some proportion of the length, not an absolute amount. Similarly I wonder how we came up with the division by 3.

I would expect something like `new ValueStringBuilder(input.Length + 50 + (int)(input.Length * 0.1));` but I suppose I'm just making that up as well.